Update libyuv to M132

youennf opened the pull request:
#### 865625bf942fce998eb842c8543a51ec950c4a1d
<pre>
Update libyuv to M132
<a href="https://bugs.webkit.org/show_bug.cgi?id=284037">https://bugs.webkit.org/show_bug.cgi?id=284037</a>
<a href="https://rdar.apple.com/140911216">rdar://140911216</a>

Reviewed by Jean-Yves Avenard.

Update to M132 and compile out some neon I8MM specific routines.

* Source/ThirdParty/libwebrtc/Configurations/libyuv.xcconfig:
* Source/ThirdParty/libwebrtc/Source/third_party/libyuv: Resynced
* Source/ThirdParty/libwebrtc/Source/third_party/libyuv/include/libyuv/row.h:
* Source/ThirdParty/libwebrtc/Source/third_party/libyuv/source/row_neon64.cc:

Canonical link: <a href="https://commits.webkit.org/287443@main">https://commits.webkit.org/287443@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f38045c5ed06d22c0325728ed37562a3129ecb37

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/79338 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/58366 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/32709 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/83953 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/30487 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/67441 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/6628 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/62058 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/19949 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/82405 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/52116 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/72285 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/42365 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/49460 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/26467 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/28885 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/70572 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/26912 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/85353 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/6622 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/4607 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/70308 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/6786 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/68159 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/69555 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/17402 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/13602 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/12468 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/6574 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/6493 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/9966 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/8289 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

jyavenard commented on Source/ThirdParty/libwebrtc/Source/third_party/libyuv/BUILD.gn:
```diff
@@ -29,10 +29,18 @@ config("libyuv_config") {
-  defines = []
+
+  # Define CHROMIUM to tell cpu_id to avoid sandbox unsafe system calls.
+  defines = [ "CHROMIUM" ]
```
do we need to do something similar here?

youennf replied:
I don;t think so, I'll compile out cpu_id.cc, this is not useful.

jyavenard commented on Source/ThirdParty/libwebrtc/Source/third_party/libyuv/BUILD.gn:
```diff
@@ -217,23 +236,41 @@ if (libyuv_use_neon) {
-
-    if (current_cpu != "arm64") {
+    if (current_cpu == "arm64") {
+      cflags = [ "-march=armv8-a+dotprod+i8mm" ]
```
is this applicable for Apple Silicon ?

jyavenard replied:
though we're aarch64 ; but for macOS it shows as arm64 so it's puzzling

youennf replied:
We compile out i8mm and dot prod specific routines from libyuv, so I don't think this matters.
@emw-apple, thoughts?

emw-apple replied:
I don't think it matters either — absent a good reason, we should stick to the default compiler settings provided by our SDKs.

youennf committed:
Update libyuv to M132
https://bugs.webkit.org/show_bug.cgi?id=284037
rdar://140911216

Reviewed by Jean-Yves Avenard.

Update to M132 and compile out some neon I8MM specific routines.

* Source/ThirdParty/libwebrtc/Configurations/libyuv.xcconfig:
* Source/ThirdParty/libwebrtc/Source/third_party/libyuv: Resynced
* Source/ThirdParty/libwebrtc/Source/third_party/libyuv/include/libyuv/row.h:
* Source/ThirdParty/libwebrtc/Source/third_party/libyuv/source/row_neon64.cc:

Canonical link: https://commits.webkit.org/287443@main
